# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1715460004 -7200
# Parent  2fa8c1d0cc8ebde11b7ab31b485d777c684ce07d
Feature: Unity menubar detached from the main window. CSS changes. Don't use global menu where it doesn't work.

There are six windows with menubar:
1) Main window 2) Compose window 3) Message stand-alone window 4) View Source 5) OpenPGP Key Manager 6) Create/Edit Calendar event.
The latter two don't work with the global menu. For the key manager the menu is hidden from the window, but the content doesn't show on the global menu.
For the calendar event creating the native menu fails in nsMenuBar::Init() because `gdkWin` is null.

diff --git a/calendar/base/content/dialogs/calendar-event-dialog.xhtml b/calendar/base/content/dialogs/calendar-event-dialog.xhtml
--- a/calendar/base/content/dialogs/calendar-event-dialog.xhtml
+++ b/calendar/base/content/dialogs/calendar-event-dialog.xhtml
@@ -235,9 +235,10 @@
            defaultlabelalign="end">
 
     <!-- Menubar -->
     <toolbar type="menubar">
-    <menubar id="event-menubar">
+    <!-- for some reason, this doesn't work with the global menu -->
+    <menubar id="event-menubar" _moz-menubarkeeplocal="true">
 
       <!-- Item menu -->
       <!-- These 2 Strings are placeholders, values are set at runtime -->
       <menu label="Item"
diff --git a/mail/extensions/openpgp/content/ui/enigmailKeyManager.xhtml b/mail/extensions/openpgp/content/ui/enigmailKeyManager.xhtml
--- a/mail/extensions/openpgp/content/ui/enigmailKeyManager.xhtml
+++ b/mail/extensions/openpgp/content/ui/enigmailKeyManager.xhtml
@@ -80,9 +80,10 @@
       <key id="key_quit" />
     </keyset>
 
     <toolbar type="menubar" style="margin-inline: -8px -10px; margin-top: -8px">
-      <menubar id="main-menubar">
+      <!-- for some reason, this doesn't work with the global menu -->
+      <menubar id="main-menubar" _moz-menubarkeeplocal="true">
         <menu id="menu_File" data-l10n-id="openpgp-key-man-file-menu">
           <menupopup id="menu_FilePopup" onpopupshowing="enigmailKeyMenu();">
             <menuitem
               id="importPubFromFile"
diff --git a/mail/themes/linux/mail/mailWindow1.css b/mail/themes/linux/mail/mailWindow1.css
--- a/mail/themes/linux/mail/mailWindow1.css
+++ b/mail/themes/linux/mail/mailWindow1.css
@@ -13,8 +13,13 @@
 @import url("chrome://messenger/skin/folderPane.css");
 @import url("chrome://messenger/skin/messageIcons.css");
 @import url("chrome://messenger/skin/shared/mailWindow1.css");
 
+/* Unity menubar detached from the main window. Show controls if no titlebar (tabsintitlebar) */
+:root[shellshowingmenubar="true"][tabsintitlebar] unified-toolbar > div > .titlebar-buttonbox-container {
+  visibility: visible !important;
+}
+
 #messengerBox {
   background-color: -moz-Dialog;
 }
 
